Mathlib/RingTheory/LaurentSeries.lean

```diff
@@ -38,6 +41,47 @@ variable {R : Type*}
 
 namespace LaurentSeries
 
+section HasseDeriv
+
+variable {V : Type*} [AddCommGroup V] [Semiring R] [Module R V]
+
+theorem hasseDeriv_bdd_below (k : ℕ) (f : LaurentSeries V) (m : ℤ) (h : m < f.order - k) :
+    (fun (n : ℤ) ↦ Ring.choose (n + k) k • HahnSeries.coeff f (n + k)) m = 0 := by
+  simp only
+  rw [@lt_sub_iff_add_lt] at h
+  rw [coeff_eq_zero_of_lt_order h, smul_zero]
+
+/-- The Laurent series given by Hasse derivative. -/
+def hasseDeriv (k : ℕ) (f : LaurentSeries V) : LaurentSeries V :=
+  HahnSeries.ofSuppBddBelow (fun (n : ℤ) => (Ring.choose (n + k) k) • f.coeff (n + k))
+  (forallLTEqZero_supp_BddBelow _ (f.order - k : ℤ) (hasseDeriv_bdd_below k f))
+theorem hasseDeriv_coeff (k : ℕ) (f : LaurentSeries V) (n : ℤ) : (hasseDeriv k f).coeff n =
+    Ring.choose (n + k) k • HahnSeries.coeff f (n + k) := rfl
```

Comment:
I claim it is better to avoid line breaks within terms in situations like this. Also we should really be consistent about use of dot notation. So I suggest:
```suggestion
theorem hasseDeriv_coeff (k : ℕ) (f : LaurentSeries V) (n : ℤ) :
    (hasseDeriv k f).coeff n = Ring.choose (n + k) k • f.coeff (n + k) :=
  rfl
```